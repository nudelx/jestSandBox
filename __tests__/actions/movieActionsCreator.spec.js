import enzymeSerializer from 'enzyme-to-json/serializer'
import {
  MovieActionsCreator,
  tryGetMovie,
  fetchingMovieProcess
} from '../../src/redux/action/movieActionCreator'

jest.mock('../../src/helper/MovieAPI')
const state = {
  movies: {},
  content: {},
  currentPage: {}
}
const dispatch = jest.fn()
const getState = jest.fn().mockImplementation(() => state)

const page = 'page_1'
const movie = { movie: 'test movie ' }
const error = 'this is a test error'

describe('it creates actions properly', () => {
  it('MovieFetching', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const state = MovieActionsCreator.startGettingMovie(page)
    expect(state).toMatchSnapshot()
  })

  it('getMovieDone', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const state = MovieActionsCreator.getMovieDone(movie, page)
    expect(state).toMatchSnapshot()
  })

  it('getMovieError', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const state = MovieActionsCreator.getMovieError(error)
    expect(state).toMatchSnapshot()
  })
})

describe('it should try to get a movie', () => {
  it('get a movie 1', () => {
    expect.assertions(1)
    return tryGetMovie(1).then(data => expect(data).toMatchSnapshot())
  })

  it('should fail to get a movie x > 3 with error message', () => {
    expect.assertions(1)
    return tryGetMovie(44).catch(err =>
      expect(err.error).toContain('44 not found')
    )
  })
})

describe('fetchingMovieProcess in action', () => {
  beforeAll(() => dispatch.mockClear())

  it('should get movie for page_1', () => {
    return fetchingMovieProcess(page)(dispatch, getState).then(() =>
      expect(dispatch.mock.calls).toMatchSnapshot()
    )
  })

  it('should get an error for page_122 not found', () => {
    dispatch.mockClear()
    expect.assertions(1)
    return fetchingMovieProcess(page + '22')(dispatch, getState).then(() =>
      expect(dispatch.mock.calls).toMatchSnapshot()
    )
  })
})
