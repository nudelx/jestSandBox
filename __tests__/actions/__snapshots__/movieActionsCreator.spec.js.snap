// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchingMovieProcess in action should get an error for page_122 not found 1`] = `
Array [
  Array [
    Object {
      "page": "page_122",
      "type": "MOVIE::FETCHING",
    },
  ],
  Array [
    Object {
      "error": Object {
        "error": "movie with 122 not found.",
      },
      "type": "MOVIE::FETCHING::ERROR",
    },
  ],
]
`;

exports[`fetchingMovieProcess in action should get movie for page_1 1`] = `
Array [
  Array [
    Object {
      "page": "page_1",
      "type": "MOVIE::FETCHING",
    },
  ],
  Array [
    Object {
      "movie": Object {
        "created": "2014-12-10 14:23:31",
        "director": "BB-8",
        "edited": "2015-04-11 09:46:52",
        "episode_id": [Circular],
        "opening_crawl": "this is a test",
        "producer": "Kylo Ren",
        "release_date": "1977-05-25",
        "title": "Hope it Will work",
        "url": "/this/is/url/to/some/movie/",
      },
      "page": "page_1",
      "type": "MOVIE::FETCHING::DONE",
    },
  ],
]
`;

exports[`it creates actions properly MovieFetching 1`] = `
Object {
  "page": "page_1",
  "type": "MOVIE::FETCHING",
}
`;

exports[`it creates actions properly getMovieDone 1`] = `
Object {
  "movie": Object {
    "movie": "test movie ",
  },
  "page": "page_1",
  "type": "MOVIE::FETCHING::DONE",
}
`;

exports[`it creates actions properly getMovieError 1`] = `
Object {
  "error": "this is a test error",
  "type": "MOVIE::FETCHING::ERROR",
}
`;

exports[`it should try to get a movie get a movie 1 1`] = `
Object {
  "created": "2014-12-10 14:23:31",
  "director": "BB-8",
  "edited": "2015-04-11 09:46:52",
  "episode_id": [Circular],
  "opening_crawl": "this is a test",
  "producer": "Kylo Ren",
  "release_date": "1977-05-25",
  "title": "Hope it Will work",
  "url": "/this/is/url/to/some/movie/",
}
`;
